simulator/src/main/java/kcl/teamIndexZero/traffic/simulator/data/trafficLight/TrafficLightController.java

```diff
@@ -1,17 +1,118 @@
 package kcl.teamIndexZero.traffic.simulator.data.trafficLight;
 
+import kcl.teamIndexZero.traffic.log.Logger;
+import kcl.teamIndexZero.traffic.log.Logger_Interface;
 import kcl.teamIndexZero.traffic.simulator.ISimulationAware;
+//import kcl.teamIndexZero.traffic.simulator.data.ID;
+import kcl.teamIndexZero.traffic.simulator.data.ID;
 import kcl.teamIndexZero.traffic.simulator.data.SimulationTick;
+import kcl.teamIndexZero.traffic.simulator.data.descriptors.TrafficLightRule;
+import kcl.teamIndexZero.traffic.simulator.data.descriptors.TrafficLightsInSetRule;
+import kcl.teamIndexZero.traffic.simulator.data.trafficLight.TrafficLightInSet;
+import kcl.teamIndexZero.traffic.simulator.data.trafficLight.TrafficLight;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
 
 /**
  * Traffic Light controller
  * Controls the traffic lights in the map based on their rules
  */
 public class TrafficLightController implements ISimulationAware {
+        public String temp;
+        public long CurrentTime;
+        public LocalDateTime Temporary;
+        public long lastChange = 0;
+        public long timer = 5;
+        public long delay;
+        public static LocalDateTime start = LocalDateTime.of(1984, 12, 16, 7, 45, 55);
+        private TrafficLight model; //list this
+        private TrafficLightInSet modelSet; //list this
+        private TrafficLightSet set;
+        public List<TrafficLight> TrafficLightSinglesList;
+        public List<TrafficLightInSet> TrafficLightSetList;
+        SimulationTick simulationTick;
+        private static Logger_Interface LOG = Logger.getLoggerInstance(TrafficLightController.class.getSimpleName());
+
+       /**
+         * Constructor
+         */
+        public TrafficLightController(){
+            TrafficLightSinglesList = new ArrayList<TrafficLight>();
+            TrafficLightSetList = new ArrayList<TrafficLightInSet>();
+        }
+
+        public long formatTimeToLong(LocalDateTime date) {
+            long milliseconds = ChronoUnit.MILLIS.between(start, date);
+            return milliseconds;
```

Comment:
How about `return ChronoUnit.MILLIS.between(start, date);` ? 1 line instead of 2 :)